src/v/bytes/iobuf.cc

```diff
@@ -72,10 +72,10 @@ ss::future<iobuf> read_iobuf_exactly(ss::input_stream<char>& in, size_t n) {
     });
 }
 
-ss::output_stream<char> make_iobuf_output_stream(iobuf io) {
+ss::output_stream<char> make_iobuf_output_stream(iobuf& io) {
```

Comment:
can we rename as `make_iobuf_ref_output_stream()` typically when you call make `std:::make_unique` you pass ownership to the make func